### PR TITLE
Handler

### PR DESCRIPTION
**Pull request by ZakayZ**

### CODE REVIEW

File: Handler/ExcitationHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  assert(evaporation_model_->GetPhotonEvaporation() != nullptr);
+}
+
+std::vector<G4ReactionProduct> ExcitationHandler::BreakItUp(const G4Fragment& fragment) {
+  G4FragmentVector results;
+  std::queue<G4Fragment*> evaporation_queue;
+  std::queue<G4Fragment*> photon_evaporation_queue;
+
+  /// In case A <= 1 the fragment will not perform any nucleon emission
+  auto initial_fragment_ptr = new G4Fragment(fragment);
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.cpp]: Isn't it a memory leak?

File: Handler/ExcitationHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+#include "G4Evaporation.hh"
+#include "G4PhotonEvaporation.hh"
+#include "G4StatMF.hh"
+#include "G4FermiBreakUpVI.hh"
+#include "G4NuclearLevelData.hh"
+
+#include "ExcitationHandler.h"
+
+static const size_t evaporation_threshold = 1e3;
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.cpp]: 1e3 is dimensionless. Better use 1*keV, to keep the dimension of the threshold. And each dimension value should have a dimension multiplied. Geant4 solves it with G4SystemOfUnits.hh

File: Handler/ExcitationHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+ExcitationHandler::ExcitationHandler()
+    : multi_fragmentation_model_(DefaultMultiFragmentation()),
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.cpp:34]: Needs an additional initialisation for ion data tables to let crate ions. See the AAMCC main file as an example for ion table init

File: Handler/ExcitationHandler.h [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+  ExcitationHandler(ExcitationHandler&&) = default;
+
+  ExcitationHandler& operator=(const ExcitationHandler&) = delete;
+
+  ExcitationHandler& operator=(ExcitationHandler&&) = default;
+
+  std::vector<G4ReactionProduct> BreakItUp(const G4Fragment& fragment);
+
+  /// parameters setters
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.h]: Is it a sketch of DI?

File: Handler/ExcitationHandler.cpp (hunk 1 above)
Comment (ZakayZ) [reply Handler/ExcitationHandler.cpp]: Maybe it was, now all pointers are unique

File: Handler/ExcitationHandler.cpp (hunk 3 above)
Comment (ZakayZ) [reply Handler/ExcitationHandler.cpp:34]: It is now initialised in the constructor

File: Handler/ExcitationHandler.cpp [hunk 5]
```diff
@@ -20,7 +21,7 @@
 
 #include "ExcitationHandler.h"
 
-static const size_t evaporation_threshold = 1e3;
+static const size_t evaporation_threshold = 1e3 * CLHEP::keV;
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.cpp]: Why 1e3 keV? This is too high energy for the evaporation threshold.

Comment (ZakayZ) [reply Handler/ExcitationHandler.cpp]: It was iteration limiter all along...

**Comment by alexsvetlichnyy** [comment]
Am I right that one can download handler feature branch and test it on my own?

File: Handler/ExcitationHandler.cpp [hunk 6]
```diff
@@ -205,7 +205,7 @@ void ExcitationHandler::ApplyFermiBreakUp(G4SmartFragment fragment,
   fermi_break_up_model_->BreakFragment(&fragments, fragment.get());
   // auto fragments = std::unique_ptr<G4FragmentVector>(fermi_break_up_model_->BreakItUp(fragment.get()))
 
-  if (fragments.size() <= 1) {
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.cpp:176]: Why it was <=1?

File: Handler/ExcitationHandler.cpp [hunk 7]
```diff
@@ -205,7 +205,7 @@ void ExcitationHandler::ApplyFermiBreakUp(G4SmartFragment fragment,
...
+  if (fragments.size() == 1) {
```
Comment (alexsvetlichnyy) [inline Handler/ExcitationHandler.cpp:208]: if fragments is empty what will be?

File: Visualizer.py
Comment (alexsvetlichnyy) [inline Visualizer.py:1]: Everything in one repo is a bad habbit

**Pull request merged**